benchmarks/htap/lib/helpers.py

```diff
@@ -284,24 +298,16 @@ def random_phone_number(self, key):
 
 
 class TimestampGenerator:
-    def __init__(self, start_date, scale_factor, random):
+    def __init__(self, start_date, random, workload_scalar = 1.0):
         self.random = random
         self.current = start_date
-        self.increment = self.timestamp_distance(scale_factor)
-
-    @staticmethod
-    def timestamp_distance(scale_factor):
-        tpch_num_orders_per_sf = 1500000
-        tpch_data_duration = isoparse('1998-08-02') - isoparse('1992-01-01')
-        tpcc_inverse_order_frac = 1 / 0.44
-        seconds_per_transaction = (
-                tpch_data_duration / (tpcc_inverse_order_frac * tpch_num_orders_per_sf)
-        )
 
-        return seconds_per_transaction / scale_factor
+        orders_per_warehouse = CUST_PER_DIST * DIST_PER_WARE
+        date_range = TPCH_DATE_RANGE[1] - TPCH_DATE_RANGE[0]
+        self.increment = (date_range / orders_per_warehouse) * workload_scalar
 
-    def next(self):
-        self.current += self.increment * self.random.gaussian(mean=1, variance=0.05)
+    def next(self, scale = 1.0):
+        self.current += self.increment # * self.random.gaussian(mean=1, variance=0.05)
```

Comment:
Either remove the comment here if it is leftover, or enable code.

Reply:
done